packages/wrangler/src/pages/upload.tsx

```diff
@@ -112,6 +112,14 @@ export const upload = async (
 
 	const directory = resolve(args.directory);
 
+	// TODO(future): Use this to more efficiently load files in and speed up uploading
```

Comment:
We can buffer the memory we have available with files we're next to upload. We upload 3 buckets at a time but with the memory available we could probably get the payload ready for the next 10 or so. Meaning that we aren't waiting on the I/O for file read each upload, we can just upload right away

Not sure how much of an improvement it would be but waiting on I/O of potentially lots of files before we can do the upload isn't fun.